lib/publiccloud/instance.pm

```diff
@@ -474,10 +479,13 @@ sub wait_for_ssh {
             script_run("ssh-keyscan $self->{public_ip} | tee ~/.ssh/known_hosts /home/$testapi::username/.ssh/known_hosts");
         }
 
+        my $exit_ssh;
         # Finally make sure that SSH works
         while (($duration = time() - $start_time) < $args{timeout}) {
-            $exit_code = $self->ssh_script_run(cmd => "true", username => $args{username}, timeout => $args{timeout} - $duration);
-            last if isok($exit_code);
+            # After the instance is resumed from hibernation the SSH can freeze
+            my $ssh_opts = $self->ssh_opts() . ' -o ControlPath=none -o ConnectTimeout=10';
+            $exit_ssh = script_run("ssh $ssh_opts -l $args{username} $self->{public_ip} true", timeout => $args{timeout} - $duration);
```

Comment:
What if instead we mantain omogeneous the code by continue using the same function already used here around? Like:

```suggestion
            $exit_ssh = $self->ssh_script_run(cmd => "true", ssh_opts => $ssh_opts, username => $args{username}, timeout => $args{timeout} - $duration, proceed_on_failure => 1);
```